src/main/java/edu/harvard/iq/dataverse/GuestbookPage.java

```diff
@@ -143,96 +118,71 @@ public String init() {
             throw new RuntimeException("On Guestook page without id or ownerid."); // improve error handling
         }
         
-        return null;
+        initCustomQuestionsForView();
         
-    }
-
-    public String removeCustomQuestion(Long index){
-        guestbook.removeCustomQuestion(index.intValue());
-        return "";
-    }
-    
-    public List<GuestbookResponse> getGuestbookResponses(){
         return null;
-    }
-    
-    private void initCustomQuestion(){
-        CustomQuestion toAdd = new CustomQuestion();
-        toAdd.setQuestionType("text");
-        toAdd.setCustomQuestionValues(new ArrayList<CustomQuestionValue>());
-        toAdd.setGuestbook(guestbook);       
-        int index = guestbook.getCustomQuestions().size();
-        guestbook.addCustomQuestion(index, toAdd);       
+        
     }
         
     public void addCustomQuestion(Integer indexIn){
         CustomQuestion toAdd = new CustomQuestion();
         toAdd.setQuestionType("text");
         toAdd.setCustomQuestionValues(new ArrayList<CustomQuestionValue>());
         toAdd.setGuestbook(guestbook);       
+        
+        addCustomQuestionValue(toAdd, 0);
+        
         guestbook.addCustomQuestion(indexIn, toAdd);
     }
+
+    public String removeCustomQuestion(Long index){
+        guestbook.removeCustomQuestion(index.intValue());
+        return "";
+    }
+    
     
     public void addCustomQuestionValue(CustomQuestion cq, int index){
         CustomQuestionValue toAdd = new CustomQuestionValue();
         toAdd.setValueString("");
         toAdd.setCustomQuestion(cq);
-        cq.addCustomQuestionValue(index, toAdd);    
+        cq.addCustomQuestionValue(index, toAdd);
     }
     
     public void removeCustomQuestionValue(CustomQuestion cq, Long index){
         cq.removeCustomQuestionValue(index.intValue());
     }
     
-    public void toggleQuestionType(CustomQuestion questionIn) {
-        if (questionIn.getCustomQuestionValues() != null && questionIn.getCustomQuestionValues().isEmpty() 
-                && questionIn.getQuestionType() !=null && questionIn.getQuestionType().equals("options")){
-            questionIn.setCustomQuestionValues(new ArrayList<CustomQuestionValue>());
-            CustomQuestionValue addCQV = new CustomQuestionValue();
-            addCQV.setCustomQuestion(questionIn);
-            questionIn.getCustomQuestionValues().add(addCQV);
-        } 
-    }
-    
-
-
-    public void edit(GuestbookPage.EditMode editMode) {
-        this.editMode = editMode;
-    }
 
     public String save() {
         boolean create = false;
+        
+        if (StringUtils.isEmpty(guestbook.getName())) {
+            FacesContext.getCurrentInstance().validationFailed();
+            FacesContext.getCurrentInstance().addMessage(guestbookName.getClientId(),
+                    new FacesMessage(SEVERITY_ERROR, null, BundleUtil.getStringFromBundle("guestbook.name.empty")));
+            return null;
```

Comment:
We can use StringUtils.Empty instead of null

Reply:
ok